Tests/NIOFileSystemIntegrationTests/FileHandleTests.swift

```diff
@@ -1178,6 +1178,121 @@ final class FileHandleTests: XCTestCase {
             }
         }
     }
+
```

Comment:
Can you also add tests to:
- `FileSystemErrorTests` to check the mapping of `errno`s to error codes?
- `SyscallTests` after you've added the appropriate function to `Syscall`